src/OpenAL/OpenToolkit.OpenAL/ALC/ALC.cs

```diff
@@ -38,22 +38,51 @@ private ALC()
         // ALC_API ALCcontext *    ALC_APIENTRY alcCreateContext( ALCdevice *device, const ALCint* attrlist );
 
         /// <summary>This function creates a context using a specified device.</summary>
-        /// <param name="device">a pointer to a device.</param>
-        /// <param name="attrlist">an array of a set of attributes: ALC_FREQUENCY, ALC_MONO_SOURCES, ALC_REFRESH, ALC_STEREO_SOURCES, ALC_SYNC.</param>
+        /// <param name="device">A pointer to a device.</param>
+        /// <param name="attributeList">A zero terminated array of a set of attributes: ALC_FREQUENCY, ALC_MONO_SOURCES, ALC_REFRESH, ALC_STEREO_SOURCES, ALC_SYNC.</param>
+        /// <returns>Returns a pointer to the new context (NULL on failure).</returns>
+        /// <remarks>The attribute list can be NULL, or a zero terminated list of integer pairs composed of valid ALC attribute tokens and requested values.</remarks>
+        [DllImport(Lib, EntryPoint = "alcCreateContext", ExactSpelling = true, CallingConvention = AlcCalliningConv)]
+        public static extern ALContext CreateContext([In] ALDevice device, [In] ref int attributeList);
+        // ALC_API ALCcontext *    ALC_APIENTRY alcCreateContext( ALCdevice *device, const ALCint* attrlist );
+
+        /// <summary>This function creates a context using a specified device.</summary>
+        /// <param name="device">A pointer to a device.</param>
+        /// <param name="attributeList">A zero terminated array of a set of attributes: ALC_FREQUENCY, ALC_MONO_SOURCES, ALC_REFRESH, ALC_STEREO_SOURCES, ALC_SYNC.</param>
         /// <returns>Returns a pointer to the new context (NULL on failure).</returns>
         /// <remarks>The attribute list can be NULL, or a zero terminated list of integer pairs composed of valid ALC attribute tokens and requested values.</remarks>
         [DllImport(Lib, EntryPoint = "alcCreateContext", ExactSpelling = true, CallingConvention = AlcCalliningConv)]
-        public static extern ALContext CreateContext([In] ALDevice device, [In] int[] attrlist);
+        public static extern ALContext CreateContext([In] ALDevice device, [In] int[] attributeList);
         // ALC_API ALCcontext *    ALC_APIENTRY alcCreateContext( ALCdevice *device, const ALCint* attrlist );
 
+        /// <summary>This function creates a context using a specified device.</summary>
+        /// <param name="device">A pointer to a device.</param>
+        /// <param name="attributeList">A zero terminated span of a set of attributes: ALC_FREQUENCY, ALC_MONO_SOURCES, ALC_REFRESH, ALC_STEREO_SOURCES, ALC_SYNC.</param>
+        /// <returns>Returns a pointer to the new context (NULL on failure).</returns>
+        /// <remarks>The attribute list can be NULL, or a zero terminated list of integer pairs composed of valid ALC attribute tokens and requested values.</remarks>
+        public static ALContext CreateContext(ALDevice device, Span<int> attributeList)
```

Comment:
Hmm should we validate the parameters? 

```cs
_ = device ?? throw new ArgumentNullException(nameof(device));
return CreateContext(device, ...);
```

Reply:
For all public facing methods, which are not direct passthrough calls for openal...

For `CreateContext`, `GetAttributeArray`, ... and so forth

Reply:
I think this method should behave like just a normal AL call and generate an AL error just like normal AL calls